libert_util/src/util.c

```diff
@@ -3134,6 +3134,35 @@ bool util_is_first_day_in_month_utc( time_t t) {
 }
 
 
+/*
+  Expects date in the order YYYY-MM-DD. The advantage of using plain
+  sscanf() instead of strptime() is that one function can be used for
+  different (one character!!) separators like:
+
+    2009-05-31 and  2018.11.01
+
+  The validator will check that the two separation characters are the
+  same.
+*/
+
+bool util_sscanf_isodate(const char * date_token , time_t * t) {
+  int day   , month , year;
+  char sep1 , sep2;
+
+  if (sscanf(date_token , "%d%c%d%c%d" , &year , &sep1 , &month , &sep2 , &day) == 5) {
```

Comment:
This should probably be: `%4d%c%2d%c%2d` where the only allowed separators are: `-` and nothing for dates. https://no.wikipedia.org/wiki/ISO_8601